### PR TITLE
Shift error into commons and fix some ide wrarnings

### PR DESCRIPTION
**Pull request by theodorekwok**
Minor changes according to our dg and fix some ide warnings. The structure shld be finalised by now.

Increase specificity of error messages for edit status and removed polling for past returns as its optional and current implementation does not allow user to clear past returns

Fix the abort in main

### CODE REVIEW

File: src/main/java/seedu/mtracker/MTracker.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public void run() {
         String[] commandComponents;
 
         do {
-            userInput = parser.getUserInput(MAIN_WORKSPACE);
+            userInput = InputParser.getUserInput(MAIN_WORKSPACE);
```
Comment (williamlamjy) [inline src/main/java/seedu/mtracker/MTracker.java:42]: is this supposed to be InputParser or parser

**Review by williamlamjy — COMMENTED** [review]
LGTM just one comment

File: src/main/java/seedu/mtracker/MTracker.java (hunk 1 above)
Comment (theodorekwok) [reply src/main/java/seedu/mtracker/MTracker.java:42]: InputParser cause getUserInput is static

**Comment by williamlamjy** [comment]
LGTM

**Pull request merged**